lib/OpenQA/Scheduler.pm

```diff
@@ -634,23 +634,54 @@ sub job_grab {
         last;
     }
 
-    my $job_hashref;
-    if ($result != 0) {
-        $job_hashref = _job_get(
-            {
-                'me.id' => schema->resultset("Jobs")->search(
-                    {
-                        state => OpenQA::Schema::Result::Jobs::RUNNING,
-                        worker_id => $workerid,
-                    }
-                )->single->id,
-            }
-        );
-        worker_set_property($workerid, 'WORKER_IP', $workerip) if $workerip;
-    }
+    my $job = schema->resultset("Jobs")->search(
+        {
+            state => OpenQA::Schema::Result::Jobs::RUNNING,
+            worker_id => $workerid,
+        }
+    )->single;
+    return {} unless ($job);
+
+    # generate jobtoken for test access
+    my @chars = ("A".."Z", "a".."z", '0'..'9');
+    my $token;
+    $token .= $chars[rand @chars] for 1..32;
```

Comment:
there's a rndstr function already in lib/db_helpers.pm resp lib/OpenQA/Plugin/Helpers.pm